cmd/apps/helm_operator_app.go

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) arkade author(s) 2020. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package apps
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path"
+
+	"github.com/alexellis/arkade/pkg/apps"
+	"github.com/alexellis/arkade/pkg/config"
+	"github.com/alexellis/arkade/pkg/env"
+	"github.com/alexellis/arkade/pkg/helm"
+	"github.com/alexellis/arkade/pkg/k8s"
+	"github.com/alexellis/arkade/pkg/types"
+	"github.com/spf13/cobra"
+	"golang.org/x/mod/semver"
+)
+
+func MakeInstallHelmOperator() *cobra.Command {
+	var helmOperator = &cobra.Command{
+		Use:          "helm-operator",
+		Short:        "Install helm-operator",
+		Long:         "Install helm-operator",
+		Example:      "arkade install helm-operator --namespace default",
+		SilenceUsage: true,
+	}
+
+	helmOperator.Flags().StringP("namespace", "n", "default", "The namespace used for installation")
+	helmOperator.Flags().Bool("update-repo", true, "Update the helm repo")
+	helmOperator.Flags().Bool("helm3", true, "Use helm3, if set to false uses helm2")
+	helmOperator.Flags().StringP("version", "v", "v1.2.0", "The version of helm-operator to install")
+
+	helmOperator.RunE = func(command *cobra.Command, args []string) error {
+		wait, _ := command.Flags().GetBool("wait")
+		namespace, _ := command.Flags().GetString("namespace")
+		version, _ := command.Flags().GetString("version")
+		updateRepo, _ := helmOperator.Flags().GetBool("update-repo")
+		kubeConfigPath, _ := command.Flags().GetString("kubeconfig")
+
+		if !semver.IsValid(version) {
+			return fmt.Errorf("%q is not a valid semver version", version)
+		}
+
+		if namespace != "default" {
+			return fmt.Errorf(`to override the "default", install via tiller`)
+		}
+
+		userPath, err := config.InitUserDir()
+		if err != nil {
+			return err
+		}
+
+		clientArch, clientOS := env.GetClientArch()
+
+		fmt.Printf("Client: %s, %s\n", clientArch, clientOS)
+
+		log.Printf("User dir established as: %s\n", userPath)
+
+		os.Setenv("HELM_HOME", path.Join(userPath, ".helm"))
+
+		_, err = helm.TryDownloadHelm(userPath, clientArch, clientOS)
+		if err != nil {
+			return err
+		}
+
+		helmRepo := "fluxcd/helm-operator"
+
+		err = helm.FetchChart(helmRepo, defaultVersion)
+		if err != nil {
+			return err
+		}
+
+		overrides := map[string]string{}
+		overrides["helm.versions"] = "v3"
```

Comment:
This flag, looking at the helm operator readme, states if the helm operator install should allow helm2&3 or just helm3 charts to be installed (by the running operator)

We should either leave it out, or have a flag like `--helm3-only` that's false by default, and not setting this override. 

https://github.com/fluxcd/helm-operator/tree/master/chart/helm-operator#installation

Reply:
not sure what I should do here since arkade only supports helm3 from now?

Reply:
good point

The readme states 

```
Helm versions supported by this operator instance, if v2 is specified then Tiller is required
```

We should probably just set it to v3, then later on (after merging a user's --set flags) check if they have put something that contains v2 and then print something along the lines of `you have specified helm2 support for the operator, please ensure tiller is installed on your cluster` 

Or you could look at doing a check for the tiller pod before trying to install? 

Im not sure what happens if you try to use `v2` without tiller pre-installed. can you check and come up with a plan?